### PR TITLE
Added option to treat all duplicate entries in the same way

### LINKED ISSUE #10372: entry merge UI: add option for treating all entries in the same way
**Is your suggestion for improvement related to a problem? Please describe.**
I'm always frustrated when I copy&merge many entries at once. It takes too long (performance issue)

**Describe the solution you'd like**
Say if I want to keep the old entry for all the new entries pasted into the library, I would love to set a check mark that indicates that all the old entries should be kept.

Of course, such a `treat the other entries like this one` check mark should also be useable for other options, like merge, keep left, keep right, etc.

### PR DESCRIPTION
**Pull request by jlauber18**
Fixes #10372

Added the checkbox "Apply to all entries" to treat all duplicate entries the same way when copy&pasted in the library.

jabref_1

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by ThiloteE** [comment]
Thank you for this pull-request.
 Looking at this from a UI design perspective, I am not completely satisfied with your proposal.

1. "Apply to all entries" raises the question "apply what to all entries?". The wording is not very self explanatory. It is short though and that's good.
2. "Apply to all entries" is located next to other ui elements that have nothing to do with "Apply to all entries".

My proposals to make this better (if technically possible):

1. Tooltip (when hovering with the mouse over the text) that additionally explains: "Treat all other entries like this one." or "Treat all other entries similar".
2. Move the checkbox above "Keep old entry".

**Comment by HoussemNasri** [comment]
I like the idea but I wonder if there is a better way to do it without a checkbox. I'm not a fan of the concept of checkboxes in the toolbar. It doesn't look good and in this case, the "Apply to all entries" option doesn't make sense when you try to merge only two entries. It's not scalable either since adding a new checkbox won't fit nicely into the current design. 

I propose something like in IntelliJ, where you have a menu button, when clicked, it display many checkboxes menus that you can check or uncheck. We can enable the "Apply to all entries" option only when user is resolving duplicates.


image

File: src/main/java/org/jabref/gui/externalfiles/ImportHandler.java [hunk 1]
```diff
@@ -201,21 +201,61 @@ public void importEntryWithDuplicateCheck(BibDatabaseContext bibDatabaseContext,
...
             DuplicateResolverDialog dialog = new DuplicateResolverDialog(existingDuplicateInLibrary.get(), entryToInsert, DuplicateResolverDialog.DuplicateResolverType.IMPORT_CHECK, bibDatabaseContext, stateManager, dialogService, preferencesService);
-            switch (dialogService.showCustomDialogAndWait(dialog).orElse(DuplicateResolverDialog.DuplicateResolverResult.BREAK)) {
+
+            DuplicateResolverDialog.DuplicateResolverResult dialogResult;
+
+            int allEntriesCase = preferencesService.getEntryResolverResult();
+            int entriesQuantity = preferencesService.getEntryPasteQuantity();
+
+            if (preferencesService.getGuiPreferences().mergeApplyToAllEntriesProperty().get()) {
+                 switch (allEntriesCase) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/externalfiles/ImportHandler.java:211]: A better idea would be to store the string value in the preferences and parse it back to the enum value using valueOf.  Makes your code much simpler.

You can look at the GroupViewMode implementation in the prefs, eg..
https://github.com/JabRef/jabref/blob/1cb4e160af80bf7af9339c4cf8d90f388bfa3de3/src/main/java/org/jabref/preferences/JabRefPreferences.java#L1406-L1411

**Comment by jlauber18** [comment]
Hello everyone, thanks for the help. Where should I put the checkbox? Or should I make it a menu button and where to put it?

**Comment by Siedlerchr** [comment]
I agree with the idea of @HoussemNasri  A menu button would be nice:
=> Create a new MenuButton "Options"
* add "only show changed fields"  as 
* add "Apply selected action to all entries" 
Just add the button next to the radio buttons. 
 

Codewise the ActionFactory has methods for creating such buttons and items. 

Example from the keybindings dialog to see how it could look:

grafik
 
CheckMenuItem:
grafik

**Comment by Siedlerchr** [comment]
@jlauber18  Any updates here?  Are you stuck? Do you need any help?

**Comment by jlauber18** [comment]
Hello @Siedlerchr , I tried to make the menu button work, but I wasn't successful. This was tagged as a Good First Issue, but it turned out to be much more complicated than I anticipated. If you think it's better to close the pull request and let someone more experienced in the project handle the issue, please feel free to do so.

**Comment by Siedlerchr** [comment]
Okay no worries, your code so far was already good. So someone of us with a bit more UI experience will try to finish up with the menu

**Comment by ThiloteE** [comment]
@jlauber18 You did a great job with your original proposal! We at JabRef should have thought a little bit more about and refined the issue, before somebody tried to implement the suggestions. I am sorry, for changing the goalposts after you had shot your shot!

**Comment by koppor** [comment]
We did not see any activity the last weeks, thus we closed the PR.

**Pull request closed without merging**